mock/py/mockbuild/backend.py

```diff
@@ -298,6 +338,14 @@ def build(self, srpm, timeout, check=True, spec=None):
                 raise Error('DynamicBuildRequires are found but support is disabled.'
                             ' See "dynamic_buildrequires" in config_opts.')
 
+            external_deps = [i for i in requires if i.startswith('external:')]
+            if external_deps:
+                if self.config.get('external_buildrequires'):
+                    self.install_external_deps(external_deps)
```

Comment:
I think this if-in-if condition would better be reversed, aka do nothing if  `external_buildrequires` is False.